### PR TITLE
Late policies fix

### PR DESCRIPTION
**Pull request by johnbumgardner**
Try to get late policies to actually apply to the assignment when they're in use

### CODE REVIEW

File: app/views/grades/_teams.html.erb [hunk 1]
```diff
@@ -32,7 +32,7 @@
 </thead>
 <!--E1877: tbody tag added to the table structure-->
 <tbody>
-<% 0.upto(@scores[:teams].length - 1).each do |index| %>
```
Comment (nnhimes) [inline app/views/grades/_teams.html.erb:35]: Does this fix #2258 ?

Comment (johnbumgardner) [reply app/views/grades/_teams.html.erb:35]: It's not ready to merge. I still need to apply the late calculation to the score

**Comment by johnbumgardner** [comment]
image
how points deductions from late submissions will be shown to the user

File: app/helpers/penalty_helper.rb [hunk 2]
```diff
@@ -3,35 +3,29 @@ def calculate_penalty(participant_id)
...
     @review_deadline_type_id = 2
     @meta_review_deadline_type_id = 5
-
     @participant = AssignmentParticipant.find(participant_id)
     @assignment = @participant.assignment
     if @assignment.late_policy_id
-      @penalty_per_unit = LatePolicy.find(@assignment.late_policy_id).penalty_per_unit
-      @max_penalty_for_no_submission = LatePolicy.find(@assignment.late_policy_id).max_penalty
-      @penalty_unit = LatePolicy.find(@assignment.late_policy_id).penalty_unit
+      puts 'got here'
```
Comment (nnhimes) [inline app/helpers/penalty_helper.rb:9]: Remove

File: app/helpers/penalty_helper.rb [hunk 3]
```diff
@@ -3,35 +3,29 @@ def calculate_penalty(participant_id)
...
+    penalties[:meta_review] = calculate_meta_review_penalty
     penalties
   end
 
   def calculate_submission_penalty
     submission_due_date = AssignmentDueDate.where(deadline_type_id: @submission_deadline_type_id,
                                                   parent_id: @assignment.id).first.due_at
-    resubmission_times = @participant.resubmission_times
+    submission_records = SubmissionRecord.where(team_id: @participant.team.id, assignment_id: @participant.assignment.id)
+    resubmission_times = submission_records.select { |submission_record| submission_record.updated_at > submission_due_date }
```
Comment (nnhimes) [inline app/helpers/penalty_helper.rb:26]: Rename to late_submission_records or something?

**Review by nnhimes — APPROVED** [review]
Good besides small comments

**Pull request merged**